### ISSUE TITLE
injectin `Auth` require is not defined

### ISSUE THREAD

**Issue by royipressburger** [ronzeidman/ng2-ui-auth#13]
Hi.

First of all thanks for the lib :)

So Im workign with ionic 2 and this is what i get when I inject 'Auth`:
(working with ionic 2 and webpack)

`ng2-ui-auth.js:1278 Uncaught ReferenceError: require is not defined(anonymous function) 
@ ng2-ui-auth.js:1278(anonymous function)
@ ng2-ui-auth.js:1(anonymous function)
@ ng2-ui-auth.js:1277__webpack_require__ 
@ bootstrap 0a4995e…:19(anonymous function) 
@ authenticationApi.ts:2__webpack_require__ 
@ bootstrap 0a4995e…:19(anonymous function) 
@ authenticationService.ts:3__webpack_require__ 
@ bootstrap 0a4995e…:19(anonymous function) 
@ login.ts:6__webpack_require__ 
@ bootstrap 0a4995e…:19(anonymous function) 
@ app.ts:3__webpack_require__ 
@ bootstrap 0a4995e…:19(anonymous function) 
@ bootstrap 0a4995e…:39__webpack_require__ 
@ bootstrap 0a4995e…:19(anonymous function) 
@ bootstrap 0a4995e…:39(anonymous function) 
@ bootstrap 0a4995e…:39`

**Comment by royipressburger** [comment]
Ok it was a problem with the webpack file:
I have added to the `noParse` : `/ng2-ui-auth/`
to avoid this: 
`WARNING in ./~/ng2-ui-auth/dist/ng2-ui-auth.min.js.map`
`Module parse failed: /ng2-ui-auth/dist/ng2-ui-auth.min.js.map Line 1: Unexpected token :
You may need an appropriate loader to handle this file type.`

Any idea how to solve this?

**Comment by ronzeidman** [comment]
It says "'require' is not defined" are you treating this mission as a common js module? It should be.

**Comment by royipressburger** [comment]
I think so.. im my `tsconfig.json` file I have `"module": "commonjs",` under compiler options

**Comment by ronzeidman** [comment]
It seems to me like a webpack configuration issue. I'll try to reproduce it tomorrow.

**Comment by ronzeidman** [comment]
https://github.com/webpack/webpack/issues/700 maybe it can help you. can you show me your webpack config file? (I couldn't reproduce the issue)

**Comment by royipressburger** [comment]
Im using the default webpack file from ionic 2 base app:
https://github.com/driftyco/ionic2-app-base/blob/master/webpack.config.js

**Comment by ronzeidman** [comment]
try to add node_modules/ng2-ui-auth to the strip-sourcemap loader

**Comment by royipressburger** [comment]
Still the same error.. Again i was able to avoid this error by removing the `noParse` : `/ng2-ui-auth/`
from the webpack config but this added :

`WARNING in ./~/ng2-ui-auth/dist/ng2-ui-auth.min.js
Critical dependencies:
1:2103-2110 require function is used in a way in which dependencies cannot be statically extracted`

**Comment by ronzeidman** [comment]
I think we're getting somewhere :) can you try to manually modify the bundled file so that you put the "require" statements at the beginning and this function (that's at the end): 
```js
module.exports=a(require("angular2/core.js"),require("angular2/http.js"),require("rxjs/Observable.js"),require("rxjs/add/observable/fromEvent.js"),require("rxjs/add/observable/interval.js"),require("rxjs/add/operator/concatMap.js"),require("rxjs/add/operator/map.js"),require("rxjs/add/operator/take.js"),require("rxjs/add/operator/takeWhile.js"))
```
will become something like this
```js
var _r1 = require("angular2/core.js");
[....]
module.exports=a(_r1,_r2,_r3,_r4,...)
```
if it fixes it maybe I'll include the modified file in the "dist" folder in the meantime and try to fix the bundling process.
also publish the webpack configuration, or add them to your "readme" pull request

**Comment by royipressburger** [comment]
still the same require warning: 
`  var _r1 = require("angular2/core");
  var _r2 = require("angular2/http");
  var _r3 = require("rxjs/Observable");
  var _r4 = require("rxjs/add/operator/map");
  var _r5 = require("rxjs/add/operator/mergeMap");
  var _r6 = require("rxjs/add/observable/interval");
  var _r7 = require("rxjs/add/observable/fromEvent");
  var _r8 = require("rxjs/add/operator/concatMap");
  var _r9 = require("rxjs/add/operator/take");
  var _r10 = require("rxjs/add/operator/takeWhile");
  module.exports = factory(_r1, _r2, _r3, _r4, _r5, _r6, _r7 ,_r8, _r9, _r10);`

**Comment by ronzeidman** [comment]
Ok, So I'll keep it the way it is in the meantime until I'll be able to get to the bottom of it.
webpack 2 should be released soon and it may fix this issue (or introduce new ones) maybe I should wait for it.

**Comment by ronzeidman** [comment]
@royipressburger I'm trying a new bundling option - rollup. (it also saves about 20KB from the bundle!)
can you try it? 
`npm install ronzeidman/ng2-ui-auth` (so you can take the new module directly from github)
I was finally able to reproduce it (webpack is complex...) and now it seems to be working

**Comment by royipressburger** [comment]
Yes. Looking good. warning is gone

**Comment by royipressburger** [comment]
@ronzeidman Hi again, I have added a pull request and the issue is there again.. it is probably a problem with the build, used `npm run compile` did you do anything else?

**Comment by ronzeidman** [comment]
Should be working. At least in the non minified version nothing but your change was changed. Maybe you are testing with the npm version instead of the github one? I haven't updated the npm version yet.

**Comment by royipressburger** [comment]
Yes that was it :) thanks again

**Comment by ronzeidman** [comment]
Committed to npm version 2.1.0.
Thanks for the cooperation and patience.

**Comment by matthieugayon** [comment]
@royipressburger I'd be curious if you ever come up with a documented solution for integrating ng2-ui-auth within Ionic 2 :)

**Comment by ronzeidman** [comment]
@royipressburger I'd love that too :). 
I haven had the chance to play with Ionic 2 yet, although I'd like too but in the near future I have to perfect my plain angular2 and have no time for anything else. 
I'll be happy to get contributions though.
It shouldn't be that different from my (a bit outdated but still relevant with minor changes) plain example: https://github.com/ronzeidman/ng2-ui-auth-example the server-side should be the same and the UI should only have the ionic environment+components+references since this library has no actual ui elements it should be a straight-forward conversion.
You are welcome to update the ng2-ui-auth-example with an "ionic" folder in addition to the "client" one if you have time.

**Comment by royipressburger** [comment]
Hi guys.. I will try to find time to do this week.
In mean while @matthieugayon if you need something spesific ask and ill try to answer

**Comment by royipressburger** [comment]
@matthieugayon @ronzeidman Hi guys.. sorry for the delay. I have created a pull request on the example project..

**Comment by ronzeidman** [comment]
Looking good. I'll setup an environment and test it. While I'm at it I'll update the examples to angular rc.4. It will probably take a couple of days.

Thanks!

**Comment by royipressburger** [comment]
Im not sure how the current version of ionic will work with rc3 so i think its best to wait for the next ionic release with that

**Comment by ronzeidman** [comment]
Everything seems to work fine now. I've merged your pull request as-is (added npm script to compile and run, added ionic to the devDependencies) and added instructions to the README.
I've updated the standard angular2 version to rc4 to use the new router and forms, I see yours was already rc3 and it worked fine so I haven't changed it.
Thanks very much, I will use your code to learn myself how to work with the new ionic2.
